File: Day<14>/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
    private:
    int digitSum(int n)
    {
    	int sum = 0;
    	while (n)
	    {
    		sum += (n % 10);
    		n /= 10;
    	}
    	return sum;
    }
    
    public:
    int RulingPair(vector<int> arr, int n)
    {
    	unordered_map<int, int> map;
    	int ans = -1;

    	for (int i = 0; i < n; i++)
		{
    		int dSum = digitSum(arr[i]);
    		if (map[dSum] != 0)
			{
    			ans = max(ans, map[dSum] + arr[i]);
    		}
    		mp[dSum] = max(map[dSum], arr[i]);
    	}
    	return ans;
    }
};

int main()
{
	int t;
	cin>>t;
	while (t--)
	{
		int n;
		cin>>n;
		vector<int> arr(n);
		for (int i = 0; i < n; ++i)
			cin>>arr[i];
	    Solution obj;
		cout << obj.RulingPair(arr,n)<<"\n";
	}
	return 0;
}

File: Day<14>/Solution.py
class Solution:

    def digitSum(self, n):
    	sum = 0
    	while (n):
    		sum += (n % 10)
    		n //= 10
    	return sum
	
    def RulingPair(self, arr, n):
    	map = dict()
    	ans = -1
    	for i in range(n):
    		dSum = self.digitSum(arr[i])

    		if (dSum in map):
    			ans = max(ans, map[dSum] + arr[i])
    		map[dSum] = max(map.get(dSum, 0) , arr[i])
    	return ans

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        obj = Solution();
        print(obj.RulingPair(arr,n))
